### PR TITLE
fix(server): handle hanging workspace after user account deleted

### PR DESCRIPTION
**Pull request by forehalo**
fix CLOUD-74

### CODE REVIEW

**Comment by forehalo** [comment]
* **#8377** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8377?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by forehalo** [comment]
fix CLOUD-74

File: packages/backend/server/src/core/doc/job.ts [hunk 1]
```diff
@@ -73,4 +81,13 @@ export class DocStorageCronJob implements OnModuleInit {
       .gauge('updates_queue_count')
       .record(await this.db.update.count());
   }
+
+  @OnEvent('user.deleted')
+  async clearUserWorkspaces(payload: EventPayload<'user.deleted'>) {
+    const workspaces = await this.permission.getOwnedWorkspaces(payload.id);
```
Comment (darkskygit) [inline packages/backend/server/src/core/doc/job.ts]: permission records will be deleted when the user is deleted, so when this event is triggered, the owned workspace can no longer be query through this api

Comment (forehalo) [reply packages/backend/server/src/core/doc/job.ts]: I delete the fkey on user.id

**Review by EYHN — APPROVED** [review]

**Review by darkskygit — APPROVED** [review]

**Pull request merged**